velox/exec/tests/RowContainerTest.cpp

```diff
@@ -59,17 +68,59 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
     }
   }
 
+  void testExtractColumnOffsetsForOddRows(
+      RowContainer& container,
+      const std::vector<char*>& rows,
+      int column,
+      const VectorPtr& expected) {
+    auto size = rows.size();
+
+    // Extract only odd rows.
+    // Test using extractColumnOffsets API.
+    auto oddsSize = size / 2;
+    auto offsetsBuffer =
+        AlignedBuffer::allocate<vector_size_t>(oddsSize, pool_.get());
+    offsetsBuffer->setSize(oddsSize * sizeof(vector_size_t));
+    auto rawOffsetsBuffer = offsetsBuffer->asMutable<vector_size_t>();
+    for (int i = 0; i < oddsSize; i++) {
+      rawOffsetsBuffer[i] = i * 2;
+    }
+
+    auto result = BaseVector::create(expected->type(), oddsSize, pool_.get());
+    container.extractColumnAtOffsets(
+        rows.data(), offsetsBuffer, column, 0, result);
+    EXPECT_EQ(oddsSize, result->size());
+    for (vector_size_t row = 0; row < oddsSize; ++row) {
+      EXPECT_TRUE(result->equalValueAt(expected.get(), row, row * 2))
+          << "at " << row << ": expected " << expected->toString(row)
+          << ", got " << result->toString();
+    }
+  }
+
   void testExtractColumnForAllRows(
       RowContainer& container,
       const std::vector<char*>& rows,
       int column,
       const VectorPtr& expected) {
     auto size = rows.size();
 
+    // Test using extractColumn API.
     auto result = BaseVector::create(expected->type(), size, pool_.get());
     container.extractColumn(rows.data(), size, column, result);
-
     assertEqualVectors(expected, result);
+
+    auto offsetsBuffer =
```

Comment:
perhaps, add // Test using extractColumnAtOffsets API. to match the comment above?